### PR TITLE
Fixing image upload and icon customization

### PR DESCRIPTION
**Pull request by chanaka3d**
Adding the option to change the icon and upload images as API thumbnails. Also provide a method to select color and background.

Screen Shot 2019-06-10 at 2 30 00 PM
Screen Shot 2019-06-10 at 2 27 11 PM

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Listing/components/ImageGenerator/Background.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
```
Comment (menakaj) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Listing/components/ImageGenerator/Background.jsx]: License header is missing.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Listing/components/ImageGenerator/ImageGenerator.jsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
+import React, { PureComponent } from 'react';
```
Comment (menakaj) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Listing/components/ImageGenerator/ImageGenerator.jsx]: License header is missing

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/components/APIDetailsTopMenu.jsx [hunk 3]
```diff
@@ -43,55 +44,62 @@ const styles = theme => ({
...
-                </Typography>
-            </div>
+        <ApiContext.Consumer>
+            {({ api }) => (
+                <div className={classes.root}>
+                    <Link to='/apis' className={classes.backLink}>
+                        <KeyboardArrowLeft className={classes.backIcon} />
+                        <div className={classes.backText}>
+                            BACK TO <br />
+                            LISTING
```
Comment (TanyaM) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/components/APIDetailsTopMenu.jsx]: Don't we need to localize these strings?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Listing/components/ImageGenerator/Background.jsx (hunk 1 above)
Comment (TanyaM) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Listing/components/ImageGenerator/Background.jsx]: License header missing

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Listing/components/ImageGenerator/ImageGenerator.jsx (hunk 2 above)
Comment (TanyaM) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Listing/components/ImageGenerator/ImageGenerator.jsx]: License Header missing

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/components/APIDetailsTopMenu.jsx [hunk 4]
```diff
@@ -43,55 +44,62 @@ const styles = theme => ({
...
-            <VerticalDivider height={70} />
+                    <VerticalDivider height={70} />
 
-            <a target='_blank' rel='noopener noreferrer' href={storeURL} className={classes.viewInStoreLauncher}>
-                <div>
-                    <LaunchIcon />
+                    <a
+                        target='_blank'
+                        rel='noopener noreferrer'
+                        href={`${window.location.origin}/store-new/apis/${api.id}/overview`}
```
Comment (TanyaM) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/components/APIDetailsTopMenu.jsx]: appcontext (store-new) need to be read from a config file. If not will break in reverse proxy scenarios.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/components/APIDetailsTopMenu.jsx (hunk 3 above)
Comment (chanaka3d) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/components/APIDetailsTopMenu.jsx]: Yes we need to. We will do an independent run and fix the missing localization keys in one go.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/components/APIDetailsTopMenu.jsx (hunk 4 above)
Comment (chanaka3d) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/components/APIDetailsTopMenu.jsx]: We need to do this across the store-new and publisher-new apps and fix them there are many places this is hard coded.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Listing/components/ImageGenerator/Background.jsx (hunk 1 above)
Comment (chanaka3d) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Listing/components/ImageGenerator/Background.jsx]: done

Comment (chanaka3d) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Listing/components/ImageGenerator/Background.jsx]: done

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Listing/components/ImageGenerator/ImageGenerator.jsx (hunk 2 above)
Comment (chanaka3d) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Listing/components/ImageGenerator/ImageGenerator.jsx]: done

Comment (chanaka3d) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Listing/components/ImageGenerator/ImageGenerator.jsx]: done

**Pull request merged**